pkg/healthcheck/healthcheck_addons.go

```diff
@@ -0,0 +1,141 @@
+package healthcheck
+
+import (
+	"context"
+	"fmt"
+
+	l5dcharts "github.com/linkerd/linkerd2/pkg/charts/linkerd2"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// LinkerdAddOnChecks adds checks to validate the add-on components
+	LinkerdAddOnChecks CategoryID = "linkerd-addons"
+
+	// LinkerdGrafanaAddOnChecks adds checks to validate the add-on components
+	LinkerdGrafanaAddOnChecks CategoryID = "linkerd-grafana"
+)
+
+var (
+	// AddOnCategories is the list of add-on category checks
+	AddOnCategories = []CategoryID{LinkerdAddOnChecks, LinkerdGrafanaAddOnChecks}
+)
+
+func (hc *HealthChecker) addOnCategories() []category {
+	return []category{
+		{
+			id: LinkerdAddOnChecks,
+			checkers: []checker{
+				{
+					description: fmt.Sprintf("%s config map exists", k8s.AddOnsConfigMapName),
+					warning:     true,
+					check: func(context.Context) error {
+						return hc.checkIfAddOnsConfigMapExists()
+					},
+				},
+			},
+		},
+		{
+			id: LinkerdGrafanaAddOnChecks,
+			checkers: []checker{
+				{
+					description: "grafana add-on service account exists",
+					warning:     false,
+					check: func(context.Context) error {
+						if grafana, ok := hc.addOns[l5dcharts.GrafanaAddOn]; ok {
+							// check for the grafana service account
+							return hc.checkServiceAccounts([]string{grafana.(map[string]interface{})["name"].(string)}, hc.ControlPlaneNamespace, "")
+						}
+						return &SkipError{Reason: "grafana add-on not enabled"}
+					},
+				},
+				{
+					description: "grafana add-on config map exists",
+					warning:     false,
+					check: func(context.Context) error {
+						if grafana, ok := hc.addOns[l5dcharts.GrafanaAddOn]; ok {
+							// check for the grafana config-map
+							_, err := hc.kubeAPI.CoreV1().ConfigMaps(hc.ControlPlaneNamespace).Get(fmt.Sprintf("%s-config", grafana.(map[string]interface{})["name"].(string)), metav1.GetOptions{})
+							if err != nil {
+								return err
+							}
+							return nil
+						}
+						return &SkipError{Reason: "grafana add-on not enabled"}
+					},
+				},
+				{
+					description: "grafana pod is running",
+					warning:     false,
+					check: func(context.Context) error {
+						if grafana, ok := hc.addOns[l5dcharts.GrafanaAddOn]; ok {
+
+							// Get grafana add-on deployment
+							deployment, err := hc.kubeAPI.AppsV1().Deployments(hc.ControlPlaneNamespace).Get(grafana.(map[string]interface{})["name"].(string), metav1.GetOptions{})
+							if err != nil {
+								return err
+							}
+							// Get grafana pod with match labels
+							grafanaPods, err := hc.kubeAPI.CoreV1().Pods(hc.ControlPlaneNamespace).List(metav1.ListOptions{LabelSelector: labels.SelectorFromSet(deployment.Spec.Template.Labels).String()})
```

Comment:
Why not leveraging `hc.controlPlanePods` instead of hitting the k8s API again?

Reply:
Yep updated.